src/main/java/com/minecolonies/client/gui/WindowTownHall.java

```diff
@@ -221,8 +275,72 @@ public WindowTownHall(BuildingTownHall.View townHall)
         registerButton(BUTTON_CHANGE_SPEC, this::doNothing);
         registerButton(BUTTON_TOGGLE_JOB, this::toggleHiring);
 
+        registerButton("up", this::updatePriority);
+        registerButton("down", this::updatePriority);
+        registerButton("delete", this::deleteWorkOrder);
     }
 
+    /**
+     * On Button click update the priority.
+     * @param button the clicked button.
+     */
+    private void updatePriority(Button button)
+    {
+        Label idLabel = (Label)button.getParent().getChildren().get(HIDDEN_ID_POSITION);
+        int id = Integer.valueOf(idLabel.getLabelText());
+        String buttonLabel = button.getID();
+
+        for (int i = 0; i < workOrders.size(); i++)
+        {
+            WorkOrderView workOrder = workOrders.get(i);
+            if (workOrder.getId() == id)
+            {
+                if(buttonLabel.equals("up"))
+                {
+                    if (i-1 >= 0)
+                    {
+                        workOrder.setPriority(workOrders.get(i-1).getPriority());
```

Comment:
Store a reference to `workOrders.get(i-1);` instead of getting it so many times.